Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/LagCalculator.java

```diff
@@ -0,0 +1,51 @@
+package com.example.android.camera2.slowmo;
+
+import android.content.ContentValues;
+import android.util.Log;
+import java.util.ArrayList;
+
+/**
+ * Calculates lag of each key-press on the server using the key-press timestamps from server and
+ * character seen timestamp from recorded video.
+ */
+public class LagCalculator {
+  private VideoProcessor videoData;
+  private ServerHandler serverData;
+  public ArrayList<Long> lagResult = new ArrayList<Long>();
+
+  public LagCalculator(VideoProcessor videoData, ServerHandler serverData) {
+    this.videoData = videoData;
+    this.serverData = serverData;
+  }
+
+  public ArrayList<Long> calculateLag() {
+    if (serverData.getServerTimestamps().size() < 2 || videoData.getResultsOCR().size() < 2) {
+      Log.d(ContentValues.TAG, "No results to show. Insufficient server-host data.");
+      return null;
+    }
+    Log.d(ContentValues.TAG, "Video start timestamp:" + videoData.getRecordStartTime());
+    // TODO: Server sequence is hardcoded. Add functionality to receive serverSequence.
+    String serverSequence = "m";
+    long serverHostSyncOffset =
+        serverData.getServerStartTimestamp() - serverData.getHostSyncTimestamp();
+    Log.d(ContentValues.TAG, "Server-Host Sync offset:" + serverHostSyncOffset);
+
+    ArrayList<Long> serverTimestamp = serverData.getServerTimestamps();
+    ArrayList<Long> videoFrameTimestamp = videoData.getVideoFrameTimestamp();
+    ArrayList<String> resultsOCR = videoData.getResultsOCR();
+
+    // Finds the video frame with the running serverSequence text.
+    for (int j = 1; j < serverTimestamp.size(); j++) {
+      for (int i = 0; i < resultsOCR.size(); i++) {
+        if (resultsOCR.get(i).startsWith(serverSequence)) {
+          long lag = videoFrameTimestamp.get(i) - serverTimestamp.get(j) + serverHostSyncOffset;
+          lagResult.add(lag);
```

Comment:
Since this is a member variable, if you call this function twice you will get the results from both calls returned in the second call.  Please make lagResult a local variable only.

Reply:
fixed